app/assets/javascripts/preferences/panes/security.tsx

```diff
@@ -0,0 +1,13 @@
+import { observer } from 'mobx-react-lite';
+import { FunctionComponent } from 'preact';
+import { PreferencesPane } from '../components';
+import { Preferences } from '../models';
+import { TwoFactorAuthComponent } from './two-factor-auth';
+
+export const Security: FunctionComponent<{ prefs: Preferences }> = observer(
+  ({ prefs }) => (
+    <PreferencesPane>
+      <TwoFactorAuthComponent tfAuth={prefs.twoFactorAuth} />
```

Comment:
Same here, instead of reading twoFactorAuth from prefs, then passing it down to TwoFactorAuthComponent, TwoFactorAuthEnabled and TwoFactorAuthDisabled, all components who need access to it should read it from the shared state